### PR TITLE
Expand SHA-3 support

### PR DESCRIPTION
**Pull request by julek-wolfssl**
Add more support in the EVP layer as well as add signing support. The SHA-3 OID's were also added for DER algorithm identifier encoding.

### CODE REVIEW

**Review by dgarske — CHANGES_REQUESTED** [review]
Changes look good. Please resolve the Jenkins build errors:

```
./configure --enable-leantls
make[2]: warning: -jN forced in submake: disabling jobserver mode.
tests/tests_unit_test-api.o: In function `test_wolfSSL_EVP_Digest_all':
api.c:(.text+0x43c5): undefined reference to `wolfSSL_EVP_Digest'

./configure --enable-sp-math-all=small --enable-all --disable-asm --enable-smallstack --enable-32bit CFLAGS='-m32'
make[2]: warning: -jN forced in submake: disabling jobserver mode.
tests/api.c: In function ‘test_wolfSSL_RSA_sign_sha3’:
tests/api.c:38522:31: error: ‘WC_SHA3_256_DIGEST_SIZE’ undeclared (first use in this function)
     const unsigned char mHash[WC_SHA3_256_DIGEST_SIZE] = {
                               ^
tests/api.c:38522:31: note: each undeclared identifier is reported only once for each function it appears in
tests/api.c:38522:25: error: unused variable ‘mHash’ [-Werror=unused-variable]
     const unsigned char mHash[WC_SHA3_256_DIGEST_SIZE] = {
                         ^
```

**Review by dgarske — APPROVED** [review]
Over to Todd for final

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: src/ssl.c [hunk 1]
```diff
@@ -50617,6 +50685,14 @@ word32 nid2oid(int nid, int grp)
                     return CTC_SHA384wECDSA;
                 case CTC_SHA512wECDSA:
                     return CTC_SHA512wECDSA;
+                case CTC_SHA3_224wECDSA:
```
[the remark this answers is not in the document]
Comment (julek-wolfssl) [reply src/ssl.c]: Protected.

File: src/ssl.c [hunk 2]
```diff
@@ -50941,6 +51017,14 @@ int oid2nid(word32 oid, int grp)
                     return CTC_SHA384wRSA;
                 case CTC_SHA512wRSA:
                     return CTC_SHA512wRSA;
+                case CTC_SHA3_224wRSA:
```
[the remark this answers is not in the document]
Comment (julek-wolfssl) [reply src/ssl.c]: Protected.

File: src/ssl.c [hunk 3]
```diff
@@ -50953,6 +51037,14 @@ int oid2nid(word32 oid, int grp)
                     return CTC_SHA384wECDSA;
                 case CTC_SHA512wECDSA:
                     return CTC_SHA512wECDSA;
+                case CTC_SHA3_224wECDSA:
```
[the remark this answers is not in the document]
Comment (julek-wolfssl) [reply src/ssl.c]: Protected.

File: src/ssl.c [hunk 4]
```diff
@@ -50605,6 +50665,14 @@ word32 nid2oid(int nid, int grp)
                     return CTC_SHA384wRSA;
                 case CTC_SHA512wRSA:
                     return CTC_SHA512wRSA;
+                case CTC_SHA3_224wRSA:
```
[the remark this answers is not in the document]
Comment (julek-wolfssl) [reply src/ssl.c]: Protected.

File: src/ssl.c [hunk 5]
```diff
@@ -41798,6 +41846,18 @@ void* wolfSSL_GetDhAgreeCtx(WOLFSSL* ssl)
                 case WC_HASH_TYPE_SHA512:
                     sigType = CTC_SHA512wECDSA;
                     break;
+                case WC_HASH_TYPE_SHA3_224:
```
[the remark this answers is not in the document]
Comment (julek-wolfssl) [reply src/ssl.c]: Protected.

File: src/ssl.c [hunk 6]
```diff
@@ -41777,6 +41813,18 @@ void* wolfSSL_GetDhAgreeCtx(WOLFSSL* ssl)
                 case WC_HASH_TYPE_SHA512:
                     sigType = CTC_SHA512wRSA;
                     break;
+                case WC_HASH_TYPE_SHA3_224:
```
[the remark this answers is not in the document]
Comment (julek-wolfssl) [reply src/ssl.c]: Protected.

**Review by SparkiDev — APPROVED** [review]

**Pull request merged**